File: php/features.php
<?php
declare(strict_types=1);

// Dog SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DogFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DogBaseFeature();
            case "test":
                return new DogTestFeature();
            default:
                return new DogBaseFeature();
        }
    }
}

File: py/dog_sdk/features.py
# Dog SDK feature factory

from dog_sdk.feature.base_feature import DogBaseFeature
from dog_sdk.feature.test_feature import DogTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DogBaseFeature(),
        "test": lambda: DogTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
